konrad/humidity/relative_humidity.py

```diff
@@ -264,159 +270,206 @@ def __call__(self, atmosphere, **kwargs):
                 # Values read from Fig. 6 in Romps (2014).
                 x=np.array([300, 240, 200, 190, 188, 186]),
                 y=np.array([0.8, 0.6, 0.7, 1.0, 0.5, 0.1]),
-                kind='linear',
-                fill_value='extrapolate',
+                kind="linear",
+                fill_value="extrapolate",
             )
 
-        return self._rh_func(atmosphere['T'][-1, :])
+        return self._rh_func(atmosphere["T"][-1, :])
+
+
 
-    
 class PolynomialCshapedRH(RelativeHumidityModel):
-    def __init__(self, top_slope = 7.5e-5, top_peak_rh = 0.75, mid_p = 500e2, mid_rh = 0.4, low_peak_p = 940e2, low_peak_rh = 0.85, bl_slope = -2e-5):
+    def __init__(
+        self,
+        top_peak_T=None,
+        top_peak_rh=0.75,
+        freezing_pt_rh=0.4,
+        bl_top_p=940e2,
+        bl_top_rh=0.85,
+        surface_rh=0.75,
+    ):
         """
-        Defines a C-shaped polynomial model. 
-        The RH increases linearly in the boundary layer and decreases linearly above the upper-tropospheric peak. 
-        The point above the upper-tropospheric peak where the RH is half the peak is coupled to the cold-point.
-        Between the two peaks, a quadratic function is defined by the point of the two peaks and one in the mid-troposphere.
-        Default values from RCEMIP large experiment statistics.
+        Defines a C-shaped polynomial model, that depends on T in the upper troposphere.
+        The RH increases linearly in the boundary layer from the surface.
+        Between the top of the boundary layer and the freezing level (T=273.15K), the rh is a quadratic function of p,
+        defined by its values at these to points, and a zero derivative at the freezing level.
+        Above the freezing level, the rh is a quadratic function of T, defined by its values at the freezing level and
+        at a chose upper-tropospheric T-value or at the cold point (see `top_peak_T` argument), and a zero derivative
+        at the freezing level.
 
         Parameters:
-            top_slope (float): slope of the linear function above the upper-tropospheric peak.
+            top_peak_T (float): Temperature of the upper tropospheric peak. If None, coupled to the cold-point.
```

Comment:
I would still be in favour of a default of `200.0` (or whatever value works best). In my opinion, this is closer to the "fixed RH(T)" assumption. But I would be open for discussion :)

Reply:
I'm fine with anything, so I changed top_peak_T default to 200K